tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,343 @@
+                fs.materialize(environment.start_date, environment.end_date)
+
+            yield environment
+        finally:
+            offline_creator.teardown()
+            fs.teardown()
+
+
+def parametrize_e2e_test(e2e_test):
+    """
+    This decorator should be used for end-to-end tests. These tests are expected to be parameterized,
+    and receive an empty feature repo created for all supported configurations.
+
+    The decorator also ensures that sample data needed for the test is available in the relevant offline store.
+
+    Decorated tests should create and apply the objects needed by the tests, and perform any operations needed
+    (such as materialization and looking up feature values).
+
+    The decorator takes care of tearing down the feature store, as well as the sample data.
+    """
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("config", FULL_REPO_CONFIGS, ids=lambda v: str(v))
+    def inner_test(config):
+        with construct_test_environment(config) as environment:
+            e2e_test(environment)
+
+    return inner_test
+
+
+def parametrize_offline_retrieval_test(offline_retrieval_test):
+    """
+    This decorator should be used by tests that rely on the offline store. These tests are expected to be parameterized,
+    and receive an Environment object that contains a reference to a Feature Store with pre-applied
+    entities and feature views.
+
+    The decorator also ensures that sample data needed for the test is available in the relevant offline store.
+
+    Decorated tests should interact with the offline store, via the FeatureStore.get_historical_features method. They
+    may perform more operations as needed.
+
+    The decorator takes care of tearing down the feature store, as well as the sample data.
+    """
+
+    configs = vary_providers_for_offline_stores(FULL_REPO_CONFIGS)
+    configs = vary_full_feature_names(configs)
+    configs = vary_infer_event_timestamp_col(configs)
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("config", configs, ids=lambda v: str(v))
+    def inner_test(config):
+        with construct_test_environment(config, create_and_apply=True) as environment:
+            offline_retrieval_test(environment)
+
+    return inner_test
+
+
+def parametrize_online_test(online_test):
+    """
+    This decorator should be used by tests that rely on the offline store. These tests are expected to be parameterized,
+    and receive an Environment object that contains a reference to a Feature Store with pre-applied
+    entities and feature views.
+
+    The decorator also ensures that sample data needed for the test is available in the relevant offline store. This
+    data is also materialized into the online store.
+
+    The decorator takes care of tearing down the feature store, as well as the sample data.
+    """
+
+    configs = vary_providers_for_offline_stores(FULL_REPO_CONFIGS)
+    configs = vary_full_feature_names(configs)
```

Comment:
Bit of a nitpick, but I've notice we flip this `vary_full_feature_names` flag for the majority of tests. Should we consider limiting how frequently we are testing it? Presumably if it's tested during consistency tests then we don't need to test it any more elsewhere.